jscomp/runtime/typed_array.ml

```diff
@@ -17,48 +17,329 @@
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
 
+type array_buffer
+type 'a array_like (* should be shared with js_array *)
```

Comment:
maybe in the future, we should create a module `Js.Array_like.t` just for types and `Js.Array_buffer.t`?